updated deprecated makedocs call: format; clean build dir; aligned code

PaulXiCao opened the pull request:
- major change: updated makedoc call. `format = :html` is deprecated.
- minor
  - cleaning build dir by default is useful
  - aligned code 

PS: this is in reference to issue #159 .

giordano commented on docs/make.jl:
```diff
@@ -1,18 +1,16 @@
-)
+    repo    = "github.com/JuliaMath/SpecialFunctions.jl.git",
+    target  = "build",
+)
```
Please, leave the newline at the end of the line.

PaulXiCao replied:
Ok, just updated my editor's settings.

PaulXiCao replied:
appended newline to file in new commit

simonbyrne commented:
Thanks @PaulXiCao. A few points:

1) I think this information would be better incorporated into the docstrings. These could then be grouped into headings as you've done here. 
2) Whether there is a BigFloat implementation is really an implementation detail, i.e. you can write something like:
```
## Implementation details

- `Float32`/`Float64`: C standard math library (libm)
- `BigFloat`: [MPFR](https://www.mpfr.org/)
```

PaulXiCao commented:
@simonbyrne I cannot find a good workflow. The documenter.jl does not use my uncommited docstrings. Do you know any way?

simonbyrne commented:
Change to the docs directory, start julia with `julia --project`, then do `]dev ..`

PaulXiCao commented:
@simonbyrne could you give a short feedback on my commits?
I have done the following for all functions in erf.jl and ellip.jl

- added documentation
- cleaned the source code 
- added tests for every possible case (data types and each piecewise definition)

My questions

1. Any feedback? Would you like something done differently?
2. If I do the same for other files would you pull it at some point? 
3. Should I split my work into different merge requests or just keep pushing to this one?

PaulXiCao commented:
Closed by mistake... *facepalm*

simonbyrne commented on docs/src/functions_overview.md:
```diff
@@ -0,0 +1,69 @@
+
+
+# WIP not grouped
+| Function                                                      | Description                                                                                                                                                     |
```
I would suggest the following groupings:
- Airy, Bessel and Hankel functions
- Gamma and related functions (anything involving `gamma`, `beta`, `logfactorial`)
- Trigonmetric integral (`sinint`, `cosint`)
- Riemann zeta and related functions (`zeta`, `eta`)

PaulXiCao replied:
Sounds good. I'm doing the trigonometric integrals right now. Will also group the other ones but their documentation will take more time..

simonbyrne commented on docs/src/functions_overview.md:
```diff
@@ -62,6 +52,17 @@ Here the *Special Functions* are listed according to the structure of [NIST Digi
 | [`besselk(nu,z)`](@ref SpecialFunctions.besselk)              | modified [Bessel function](https://en.wikipedia.org/wiki/Bessel_function) of the second kind of order `nu` at `z`                                               |
 | [`besselkx(nu,z)`](@ref SpecialFunctions.besselkx)            | scaled modified Bessel function of the second kind of order `nu` at `z`                                                                                         |
+
+## Gamma and Related Functions
```
"Related" shouldn't be capitalized.

simonbyrne commented on docs/src/functions_overview.md:
```diff
@@ -71,3 +72,9 @@ Here the *Special Functions* are listed according to the structure of [NIST Digi
 | [`logabsbeta(x,y)`](@ref SpecialFunctions.logabsbeta)                                          | accurate `log(abs(beta(x,y)))` for large `x` or `y`     |
 | [`logabsbinomial(x,y)`](@ref SpecialFunctions.logabsbinomial)                                          | accurate `log(abs(beta(x,y)))` for large `x` or `y`     |
+
+## Riemann Zeta and Related Functions
```
Same: "Related Functions"  shouldn't be capitalized.

PaulXiCao replied:
fixed

simonbyrne commented:
Looks much better, thanks!
